### PR TITLE
Fix stop show list files when 'limit' is reached #479

### PR DESCRIPTION
**Pull request by u5surf**
#479 
I fix to stop showing file list when limit is reached.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2df95f4ec8366c435259779c12bd81729f54383c
  * Commits must be in the following format: **%{scope}: %{description}**



---
This message was auto-generated by https://gitcop.com

File: cmd/messages.go [hunk 1]
```diff
@@ -120,17 +120,21 @@ func callLsMessages(opts map[string]string) error {
...
+		"limit":  opts["limit"],
+	})
 	reader := bufio.NewReader(os.Stdin)
 	fmt.Print("next page...")
 	if _, err := reader.ReadString('\n'); err != nil {
 		return err
 	}
 
-	return callLs(map[string]string{
+	return callLsMessages(map[string]string{
```
Comment (sanderpick) [inline cmd/messages.go]: I think this is the only change needed.

File: cmd/messages.go [hunk 2]
```diff
@@ -120,17 +120,21 @@ func callLsMessages(opts map[string]string) error {
 	if err != nil {
 		return err
 	}
-	if len(list) < limit {
+	if len(list) <= limit {
 		return nil
 	}
-
+	callLs(map[string]string{
```
Comment (sanderpick) [inline cmd/messages.go]: No need to call this here.

File: cmd/messages.go [hunk 3]
```diff
@@ -120,17 +120,21 @@ func callLsMessages(opts map[string]string) error {
 	if err != nil {
 		return err
 	}
-	if len(list) < limit {
+	if len(list) <= limit {
```
Comment (sanderpick) [inline cmd/messages.go]: If this is `<=`, you won't ever get anything past the first page. We could add some kind of "more" flag to the response to indicate that there are more results waiting on the node, but that would require some extra database work... may as well just make one more call and get an empty slice in the case where the total results equal limit.

**Comment by sanderpick** [comment]
Cheers @u5surf !

**Pull request merged**